### PR TITLE
Add LLM default headers support

### PR DESCRIPTION
**Pull request by herculesggimenes**
## Summary
- Parse optional `LLM_DEFAULT_HEADERS_JSON` as a JSON object of string headers
- Apply those headers to frontend Gemini/OpenAI provider factories
- Apply those headers to app-server Gemini/OpenAI reqwest clients
- Document the optional env var for OpenAI-compatible gateways

This pairs with lmnr-ai/lmnr-helm#35, which renders the same env var from `llmDefaultHeaders`.

## Verification
- `pnpm --dir frontend exec tsx --test tests/test-llm-default-headers.test.ts`
- `pnpm --dir frontend exec tsc --noEmit --target ES2022 --module ESNext --moduleResolution Bundler --jsx preserve --types node --skipLibCheck lib/ai/model.ts`
- `pnpm --dir frontend exec eslint lib/ai/model.ts tests/test-llm-default-headers.test.ts`
- `git diff --check`

Full frontend `pnpm --dir frontend type-check` is currently blocked by pre-existing missing asset imports under `@/assets/...`. App-server cargo verification is currently blocked locally by missing `protoc`.

---

> [!NOTE]
> **Low Risk**
> Configuration-only change with strict JSON validation; no auth or data-path changes, and Bedrock is unaffected.
> 
> **Overview**
> Adds optional **`LLM_DEFAULT_HEADERS_JSON`** so deployments can send static HTTP headers on every LLM request—mainly for **OpenAI-compatible gateways** that need tenant or version headers alongside `LLM_BASE_URL`.
> 
> **Frontend:** `parseLlmDefaultHeaders` reads the env var (JSON object of strings, validated) and passes `headers` into OpenAI and Gemini AI SDK factories in `getLanguageModel`. Bedrock is unchanged.
> 
> **App-server:** Shared `default_headers_from_env` / `parse_default_headers_json` builds a `reqwest` `HeaderMap` and attaches it via `.default_headers()` on Gemini and OpenAI HTTP clients at init. Invalid config fails client creation with clear errors; unit tests cover parsing edge cases.
> 
> **Docs:** README notes server-side AI workers use the same `.env` LLM config and documents the new variable under the OpenAI gateway example.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b6fea16e8c981e57cba1289eb32f86cde065ca8b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by herculesggimenes** [comment]
@greptile-apps please re-review. I pushed 6a75433d to address the current review feedback: broadened README docs beyond OpenAI, added frontend HTTP header name/value validation, and made `isAiProviderConfigured()` return false when `LLM_DEFAULT_HEADERS_JSON` is invalid.

File: frontend/lib/ai/model.ts [hunk 1]
```diff
@@ -77,12 +106,13 @@ export function getLanguageModel(tier: ModelTier = "large"): LanguageModel {
 
   const apiKey = process.env.LLM_API_KEY;
   const baseURL = process.env.LLM_BASE_URL;
+  const headers = parseLlmDefaultHeaders();
```
[the remark this answers is not in the document]
Comment (herculesggimenes) [reply frontend/lib/ai/model.ts:128]: Fixed in 6a75433d: `isAiProviderConfigured()` now requires `hasValidLlmDefaultHeaders()`, and the focused test asserts invalid `LLM_DEFAULT_HEADERS_JSON` keeps the provider unconfigured.

File: frontend/lib/ai/model.ts [hunk 2]
```diff
@@ -58,6 +59,34 @@ function resolveModelName(provider: LLMProvider, tier: ModelTier): string {
...
+  const headers: LlmDefaultHeaders = {};
+  for (const [name, headerValue] of Object.entries(parsed)) {
+    if (typeof headerValue !== "string") {
+      throw new Error(`Invalid LLM_DEFAULT_HEADERS_JSON: header '${name}' value must be a string`);
+    }
+    headers[name] = headerValue;
+  }
+
+  return Object.keys(headers).length > 0 ? headers : undefined;
+}
```
[the remark this answers is not in the document]
Comment (herculesggimenes) [reply frontend/lib/ai/model.ts:89]: Fixed in 6a75433d: `parseLlmDefaultHeaders()` now calls `validateHeader()`, which constructs a WHATWG `Headers` object to reject invalid header names/values; tests cover spaces in names and embedded newlines.

File: frontend/lib/ai/model.ts [hunk 3]
```diff
@@ -51,13 +52,60 @@ function getConfiguredLLMProvider(): LLMProvider | null {
...
+
+function hasValidLlmDefaultHeaders(): boolean {
+  try {
+    parseLlmDefaultHeaders();
+    return true;
+  } catch {
```
Comment (Rainhunter13) [inline frontend/lib/ai/model.ts:95]: would be good to add a warning log here so that the error is clear in case of malformed headers env var

**Review by Rainhunter13 — APPROVED** [review]

File: frontend/lib/ai/model.ts (hunk 3 above)
Comment (Rainhunter13) [reply frontend/lib/ai/model.ts:95]: non-blocking, will add later
